src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ProvideStreamMemoryAsyncOverridesTests.cs

```diff
@@ -0,0 +1,1476 @@
+        private const string CSHideWriteAsyncArray = @"public Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) => throw null;";
+        private const string CSHideWriteAsyncMemory = @"public ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default) => throw null;";
+
+        private const string VBHideReadAsyncArray = @"Public Function ReadAsync(buffer() As Byte, offset As Integer, count As Integer, cancellationToken As CancellationToken) As Task(Of Integer)
+            Throw New NotImplementedException()
+        End Function";
+        private const string VBHideWriteAsyncArray = @"Public Function WriteAsync(buffer As Byte, offset As Integer, count As Integer, cancellationToken As CancellationToken) As Task
+            Throw New NotImplementedException()
+        End Function";
+
+        private const string CSHideExplicitReadAsyncArray = @"public new Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) => throw null;";
+        private const string CSHideExplicitReadAsyncMemory = @"public new ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default) => throw null;";
+        private const string CSHideExplicitWriteAsyncArray = @"public new Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) => throw null;";
+        private const string CSHideExplicitWriteAsyncMemory = @"public new ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default) => throw null;";
+
+        private const string VBHideExplicitReadAsyncArray = @"Public Overloads Function ReadAsync(buffer() As Byte, offset As Integer, count As Integer, cancellationToken As CancellationToken) As Task(Of Integer)
+            Throw New NotImplementedException()
+        End Function";
+        private const string VBHideExplicitReadAsyncMemory = @"Public Overloads Function ReadAsync(buffer As Memory(Of Byte), Optional cancellationToken As CancellationToken = Nothing) As ValueTask(Of Integer)
+            Throw New NotImplementedException()
+        End Function";
+        private const string VBHideExplicitWriteAsyncArray = @"Public Overloads Function WriteAsync(buffer As Byte, offset As Integer, count As Integer, cancellationToken As CancellationToken) As Task
+            Throw New NotImplementedException()
+        End Function";
+        private const string VBHideExplicitWriteAsyncMemory = @"Public Overloads Function WriteAsync(buffer As ReadOnlyMemory(Of Byte), Optional cancellationToken As CancellationToken = Nothing) As ValueTask
+            Throw New NotImplementedException()
+        End Function";
+
+        private static DiagnosticDescriptor Rule => ProvideStreamMemoryBasedAsyncOverrides.Rule;
+        private static string RuleId => ProvideStreamMemoryBasedAsyncOverrides.RuleId;
```

Comment:
Could thie be a constant?

```suggestion
        private const string RuleId = ProvideStreamMemoryBasedAsyncOverrides.RuleId;
```

Reply:
I suppose it could. I figured since it's just a shorthand for accessing the field on the analyzer class, it makes sense just to make it a property that simply returns the field.

Reply:
@NewellClark for the purpose of test. It probably doesn't matter a lot. But it's better practice to use private constant.

See [CA1802](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1802?view=vs-2019).

Reply:
CA1802 talks about using a static readonly field instead of a constant. A non-auto readonly property has no associated storage, and thus does not need a static constructor to initialize it at runtime.